ml_model/preprocessing/preprocess_os_cmd.py

```diff
@@ -0,0 +1,113 @@
+"""
+Copyright (C) Microsoft Corporation. All rights reserved.​
+"""
+from azureml.core.run import Run
+import argparse
+import subprocess
+from util.model_helper import get_or_register_dataset, get_aml_context
+
+
+def main():
+    print("Running preprocess_os_cmd.py")
+
+    parser = argparse.ArgumentParser("preprocess_os_cmd")
+    parser.add_argument(
+        "--dataset_name",
+        type=str,
+        help=("Dataset name. Dataset must be passed by name\
+              to always get the desired dataset version\
+              rather than the one used while the pipeline creation")
+    )
+
+    parser.add_argument(
+        "--datastore_name",
+        type=str,
+        help=("Datastore name. If none, use the default datastore")
+    )
+
+    parser.add_argument(
+        "--data_file_path",
+        type=str,
+        help=("data file path, if specified,\
+               a new version of the dataset will be registered")
+    )
+
+    parser.add_argument(
+        "--caller_run_id",
+        type=str,
+        help=("caller run id, for example ADF pipeline run id")
+    )
+
+    parser.add_argument(
+        "--step_output",
+        type=str,
+        help=("output for passing data to next step")
+    )
+
+    args = parser.parse_args()
+
+    print("Argument [dataset_name]: %s" % args.dataset_name)
+    print("Argument [datastore_name]: %s" % args.datastore_name)
+    print("Argument [data_file_path]: %s" % args.data_file_path)
+    print("Argument [caller_run_id]: %s" % args.caller_run_id)
+    print("Argument [step_output]: %s" % args.step_output)
+
+    data_file_path = args.data_file_path
+    dataset_name = args.dataset_name
+    datastore_name = args.datastore_name
+    step_output_path = args.step_output
+
+    run = Run.get_context()
+
+    # Get Azure machine learning workspace
+    aml_workspace, *_ = get_aml_context(run)
+
+    # Get the dataset
+    dataset = get_or_register_dataset(
+        dataset_name,
+        datastore_name,
+        data_file_path,
+        aml_workspace)
+
+    # Link dataset to the step run so it is trackable in the UI
+    run.input_datasets['input_dataset'] = dataset
+    run.parent.tag("dataset_id", value=dataset.id)
+
+    # Process data
+    mount_context = dataset.mount()
+    mount_context.start()
+    print(f"mount_point is: {mount_context.mount_point}")
+
+    ####
+    # Execute something here just 'cp' from input to output folder
+    #   cp /tmp/. /tmp2 -r
+    # Prepackage any command line tools needed,
+    # in the docker image (ml_model/preprocessing/Dockerfile)
+    process = subprocess.Popen(['cp',
```

Comment:
Here the tool is called. In this demo, we just call `cp` to move input to output